libraries/AP_DroneCAN/AP_DroneCAN.h

```diff
@@ -326,6 +326,9 @@ class AP_DroneCAN : public AP_CANDriver, public AP_ESC_Telem_Backend {
     Canard::ObjCallback<AP_DroneCAN, uavcan_equipment_esc_Status> esc_status_cb{this, &AP_DroneCAN::handle_ESC_status};
     Canard::Subscriber<uavcan_equipment_esc_Status> esc_status_listener{esc_status_cb, _driver_index};
 
+    Canard::ObjCallback<AP_DroneCAN, uavcan_equipment_esc_StatusExtended> esc_status_extended_cb{this, &AP_DroneCAN::handle_esc_ext_status};
```

Comment:
this should have the #if for AP_EXTENDED_ESC_TELEM_ENABLED

Reply:
also, how did this build? The function AP_DroneCAN::handle_esc_ext_status shouldn't have existed, so surprising the compiler is OK with this

Reply:
.... the body of the function is being removed, rather than the function.  That should be fixed.

Reply:
Modified code to apply this compiler directive (AP_EXTENDED_ESC_TELEM_ENABLED) for the callback function.